lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -244,20 +253,28 @@ defmodule Trento.Domain.SapSystem do
   # Restore sap system
   # Same registration rules
   def execute(
-        %SapSystem{deregistered_at: deregistered_at} = sap_system,
-        %RegisterApplicationInstance{} = instance
+        %SapSystem{
+          application: %Application{instances: instances},
+          deregistered_at: deregistered_at
+        } = sap_system,
+        %RegisterApplicationInstance{host_id: host_id, instance_number: instance_number} = command
```

Comment:
I wouldn't rename this to `command`. It was ok as `instance`

Reply:
I thought to rename it as I need to use `instance = get_instance(...)` further down